File: maximum-matrix-sum.c
long long maxMatrixSum(int** matrix, int matrixSize, int* matrixColSize) {
    long long sum = 0;
    int minVal = INT_MAX, negCount = 0, zeroCount = 0;
    for (int i = 0; i < matrixSize; i++) {
        for (int j = 0; j < matrixColSize[i]; j++) {
            int cell = matrix[i][j];
            if (cell < 0) {
                sum -= cell;
                negCount++;
                if (-cell < minVal) minVal = -cell;
            } else if (cell == 0) {
                zeroCount++;
                if (0 < minVal) minVal = 0;
            } else {
                sum += cell;
                if (cell < minVal) minVal = cell;
            }
        }
    }
    return (zeroCount > 0 || negCount % 2 == 0) ? sum : sum - 2 * minVal;
}

File: maximum-matrix-sum.java
class Solution {
    public long maxMatrixSum(int[][] matrix) {
        long sum = 0;
        int neg = 0, min = Integer.MAX_VALUE;
        for(int i = 0; i < matrix.length; i++) {
            for(int j = 0; j < matrix.length; j++) {
                int val = Math.abs(matrix[i][j]);
                if(matrix[i][j] < 0) neg++; 
                if(val < min) min = val;
                sum += val;
            }
        }
        return (neg & 1) == 0 ? sum : sum - 2 * min;
    }
}
